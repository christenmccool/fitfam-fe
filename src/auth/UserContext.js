import React from 'react';

/** Context: provide access to user object throughout app */

const UserContext = React.createContext();

export default UserContext;
