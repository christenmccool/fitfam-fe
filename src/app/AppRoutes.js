import React, {useContext} from 'react';
import { Routes, Route, Navigate } from 'react-router-dom';

import UserContext from '../auth/UserContext';
import Homepage from './Homepage';
import WorkoutSearchPage from '../workouts/WorkoutSearchPage';
import WorkoutDetail from '../workouts/WorkoutDetail';
import PostingList from '../postings/PostingList';
import PostingDetail from '../postings/PostingDetail';
import ResultFormPage from '../results/ResultFormPage';
import ResultDetail from '../results/ResultDetail';
import LoginForm from '../auth/LoginForm';
import SignupForm from '../auth/SignupForm';

/** Routes for FitFam app
 *
 * Parts of site should only be visitable when logged in. 
 *
 * Non-existant route redirects to the homepage.
 */
const AppRoutes = ({ login, signup }) => {
  const {user} = useContext(UserContext);

  const homeElement = user ? <PostingList /> : <Homepage /> ;

  return (
    <Routes>
      <Route path="/" element={homeElement} />
      <Route path="/workouts/" element={<WorkoutSearchPage />} />
      <Route path="/workouts/:id" element={<WorkoutDetail />} />
      <Route path="/postings/:id" element={<PostingDetail />} />
      <Route path="/postings/:postId/results/" element={<ResultFormPage />} />
      <Route path="/results/:id" element={<ResultDetail />} />
      <Route path="/login" element={<LoginForm login={login}/>} />
      <Route path="/signup" element={<SignupForm signup={signup}/>} />
      <Route path="*" element={<Navigate replace to="/" />} />
    </Routes> 
  )
}

export default AppRoutes;