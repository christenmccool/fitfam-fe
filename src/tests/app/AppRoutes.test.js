import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AdapterMoment from '@mui/lab/AdapterMoment';
import LocalizationProvider from '@mui/lab/LocalizationProvider';

import UserContext from "../../auth/UserContext";
import AppRoutes from "../../app/AppRoutes";

import {demoUser } from "../testutils";


it('renders without crashing when not signed in', () => {
  render(
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <UserContext.Provider value={{user: null}}>
          <AppRoutes />
        </UserContext.Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );
});


it("matches snapshot when not signed in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <UserContext.Provider value={{user: null}}>
          <AppRoutes />
        </UserContext.Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it('renders without crashing when signed in', () => {
  render(
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <UserContext.Provider value={{user: demoUser}}>
          <AppRoutes />
        </UserContext.Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );
});


it("matches snapshot when signed in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterMoment}>
        <UserContext.Provider value={{user: demoUser}}>
          <AppRoutes />
        </UserContext.Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});



