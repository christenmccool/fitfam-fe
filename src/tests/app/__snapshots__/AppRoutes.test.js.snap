// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when not signed in 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root MuiContainer-maxWidthSm css-ead4d9-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-19rm58x"
    >
      <div
        class="MuiBox-root css-1l4w6pd"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1tazx64-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-3"
            id="mui-3-label"
          >
            Date
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Choose date, selected date is Mar 25, 2022"
              aria-readonly="true"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-3"
              readonly=""
              type="text"
              value="03/25/2022"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-7nxzgd"
              >
                <span>
                  Date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        align="center"
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-19rm58x"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Loading
          </h6>
        </div>
      </div>
      <a
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-1tgvzpz-MuiButtonBase-root-MuiButton-root"
        href="/workouts"
        tabindex="0"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Search for more workouts
        </h6>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when signed in 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-19rm58x"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Loading
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
